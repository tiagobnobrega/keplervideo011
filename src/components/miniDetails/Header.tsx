import { isEqual } from 'lodash';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { COLORS } from '../../styles/Colors';
import { TitleData } from '../../types/TitleData';
import { scaleUxToDp } from '../../utils/pixelUtils';

interface HeaderProps {
  data?: TitleData;
}

const Header = ({ data }: HeaderProps) => {
  return (
    <View style={styles.headerContent}>
      <Text style={styles.titleText} numberOfLines={2}>
        {data?.title}
      </Text>
      <Text style={styles.descriptionText} numberOfLines={3}>
        {data?.description}
      </Text>
    </View>
  );
};

const headerDataUnchanged = (
  prevProps: HeaderProps,
  nextProps: HeaderProps,
) => {
  return isEqual(prevProps.data, nextProps.data);
};

export default React.memo(Header, headerDataUnchanged);

const styles = StyleSheet.create({
  headerContent: {
    paddingStart: scaleUxToDp(120),
    paddingTop: scaleUxToDp(138),
    width: scaleUxToDp(884),
  },
  titleText: {
    fontSize: scaleUxToDp(60),
    fontWeight: 'bold',
    color: COLORS.SMOKE_WHITE,
    width: scaleUxToDp(800),
  },
  descriptionText: {
    fontSize: scaleUxToDp(30),
    color: COLORS.SMOKE_WHITE,
    width: scaleUxToDp(884),
  },
});
