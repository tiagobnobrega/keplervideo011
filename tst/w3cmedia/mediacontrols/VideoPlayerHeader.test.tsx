/*
 * Copyright (c) 2024 Amazon.com, Inc. or its affiliates.  All rights reserved.
 *
 * PROPRIETARY/CONFIDENTIAL.  USE IS SUBJECT TO LICENSE TERMS.
 */
import { describe } from '@jest/globals';
import '@testing-library/jest-native/extend-expect';
import { render } from '@testing-library/react-native';
import React from 'react';
import Header from '../../../src/w3cmedia/mediacontrols/VideoPlayerHeader';

const mockNavigateBack = jest.fn();
const renderCaptionButton = () => {
  return <Header title={'Some Title'} navigateBack={mockNavigateBack} />;
};
describe('Header tests', () => {
  it('component renders correctly', () => {
    const component = render(renderCaptionButton());
    expect(component).toMatchSnapshot();
  });
});
