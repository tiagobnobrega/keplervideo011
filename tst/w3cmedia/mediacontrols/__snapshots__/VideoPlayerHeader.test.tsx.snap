// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 0.8,
      "paddingLeft": 20,
      "paddingRight": 40,
      "paddingVertical": 10,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      Object {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      Object {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hasTVPreferredFocus={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
        "padding": 25,
        "width": 125,
      }
    }
  >
    <Text
      testID="header_back_icon"
    />
  </View>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 70,
        "fontWeight": "500",
        "textAlign": "right",
        "width": "90%",
      }
    }
    testID="video-player-header"
  >
    Some Title
  </Text>
</View>
`;
